### ISSUE TITLE
Method of calculating GPU utilization when applying NVIDIA Multi-Instance GPU

### ISSUE THREAD

**Issue by Jea-Eok-Kim** [NVIDIA/gpu-monitoring-tools#151]
[NVIDIA Multi-Instance GPU User Guide :: NVIDIA Tesla Documentation](https://docs.nvidia.com/datacenter/tesla/mig-user-guide/#cuda-gi)

does not support properties of utilization metrics for MIG devices, according to Is there another way to calculate the usage rate of gpu?

**Comment by qisikai** [comment]
+1

**Comment by qisikai** [comment]
You can look at this project: https://github.com/NVIDIA/DCGM  
DCGM can get utilization metrics of mig device according to nv.

**Comment by Jea-Eok-Kim** [comment]
No. If you refer to https://github.com/NVIDIA/gpu-monitoring-tools/issues/152,

In the MIG Enable state, dcgmi dmon -e 203 (DCGM_FI_DEV_GPU_UTIL) runs the following results:

![MIG_Enabel1](https://user-images.githubusercontent.com/35481922/108959824-4fcfa380-76b8-11eb-9f7b-9dd5c0dedcb6.png)
![DCGM-1](https://user-images.githubusercontent.com/35481922/108959832-53fbc100-76b8-11eb-8c54-7e9371277779.png)

**Comment by qisikai** [comment]
You can create a new group, then add mig compute instance to the group
when you run dcgmi, you can specify the group by `-g` .

**Comment by Jea-Eok-Kim** [comment]
@qisikai

Hello, I added and ran a group through dcgmi using https://docs.nvidia.com/datacenter/dcgm/latest/dcgm-user-guide/feature-overview.html#sgroup, but I don't think it's working.

dcgmi group -c GPU_Group
dcgmi group -g 4 -a 0,1,2,3,4,5,6,7
dcgmi dmon -e 203 -g 4


![스크린샷 2021-02-24 오후 5 27 43](https://user-images.githubusercontent.com/35481922/108971847-1736c680-76c6-11eb-906e-056e224e3a08.png)

![스크린샷 2021-02-24 오후 5 29 02](https://user-images.githubusercontent.com/35481922/108971978-359cc200-76c6-11eb-9842-f0051412c6bb.png)

**Comment by qisikai** [comment]
Try this:

```
# dcgmi discovery -c
.....
| -> I 6/0          | GPU Instance (EntityID: 42)                                        |
|    -> CI 6/0/0    | Compute Instance (EntityID: 42)                                    |
|    -> CI 6/0/1    | Compute Instance (EntityID: 43)                                    |
```

`dcgmi group -g 4 -a  compute_instance:42`

**Comment by Jea-Eok-Kim** [comment]
@qisikai

Hello
The results came out as follows, so can we extract the amount of gpu usage for each mig profile?

<img width="862" alt="스크린샷 2021-02-24 오후 6 39 06" src="https://user-images.githubusercontent.com/35481922/108981522-26bb0d00-76d0-11eb-927d-bd538547e166.png">

<img width="982" alt="스크린샷 2021-02-24 오후 6 41 02" src="https://user-images.githubusercontent.com/35481922/108981535-29b5fd80-76d0-11eb-806e-a072834f0205.png">

Thank you.

**Comment by qisikai** [comment]
I think so. 
When a card is in mig mode, then dcgm is needed to capture the metrics. 
When a card is not in mig mode, then you can use `nvidia-smi` or `nvml sdk` to get  metrics.

ps: (I stuck here, dcgmi output 0 even it has load. I have contacted nvidia for help,you can have a try on your machine.)

**Comment by Jea-Eok-Kim** [comment]
@qisikai 

I checked and found that the gpu in the MIG enable state was maintaining 0% during deep learning.

The gpu in the MIG enable state seems to be fixed at 0%.


![스크린샷 2021-02-24 오후 8 59 33](https://user-images.githubusercontent.com/35481922/108998157-d352ba00-76e3-11eb-9a9c-7ff6c5e9d70a.png)

**Comment by qisikai** [comment]
@Jea-Eok-Kim Thanks，Have you tried using mig card （Ex. GPU-CI 49） to train a model and print the utilization?

**Comment by Jea-Eok-Kim** [comment]
@qisikai 

As you can see, these are the areas where you set up MIG enable, and the example code used this.

![스크린샷 2021-02-24 오후 9 32 37](https://user-images.githubusercontent.com/35481922/109001581-73aadd80-76e8-11eb-8fd7-f2fb05022ec4.png)


https://github.com/horovod/horovod/blob/master/examples/tensorflow2/tensorflow2_keras_mnist.py

**Comment by qisikai** [comment]
@Jea-Eok-Kim Thanks. If I have any progress, I will reply here.

**Comment by nikkon-dev** [comment]
Hi, for the GPU utilization in the MIG mode, DCGM supports only the DCGM_FI_PROF* group of metrics (except for NVLINK, which is naturaly disabled in MIG mode). DCGM_FI_DEV_GPU_UTIL is not and will not be supported in MIG mode.
The dcgm-exporter itself does not support MIG devices yet - this is a work in progress.
